cmd/opm/alpha/bundle/build.go

```diff
@@ -44,33 +45,41 @@ func newBundleBuildCmd() *cobra.Command {
 		RunE: buildFunc,
 	}
 
-	bundleBuildCmd.Flags().StringVarP(&dirBuildArgs, "directory", "d", "", "The directory where bundle manifests and metadata for a specific version are located")
+	bundleBuildCmd.Flags().StringVarP(&dirBuildArgs, "directory", "d", "",
+		"The directory where bundle manifests and metadata for a specific version are located " +
+		"(Can be used for interpreting bundle information if in the nested bundle format.)")
 	if err := bundleBuildCmd.MarkFlagRequired("directory"); err != nil {
 		log.Fatalf("Failed to mark `directory` flag for `build` subcommand as required")
 	}
 
-	bundleBuildCmd.Flags().StringVarP(&tagBuildArgs, "tag", "t", "", "The image tag applied to the bundle image")
+	bundleBuildCmd.Flags().StringVarP(&tagBuildArgs, "tag", "t", "",
+		"The image tag applied to the bundle image")
 	if err := bundleBuildCmd.MarkFlagRequired("tag"); err != nil {
 		log.Fatalf("Failed to mark `tag` flag for `build` subcommand as required")
 	}
 
-	bundleBuildCmd.Flags().StringVarP(&packageNameArgs, "package", "p", "", "The name of the package that bundle image belongs to")
-	if err := bundleBuildCmd.MarkFlagRequired("package"); err != nil {
-		log.Fatalf("Failed to mark `package` flag for `build` subcommand as required")
-	}
+	bundleBuildCmd.Flags().StringVarP(&packageNameArgs, "package", "p", "",
+		"The name of the package that bundle image belongs to " +
+		"(Required if `directory` is not pointing to a bundel in the nested bundel format)")
 
-	bundleBuildCmd.Flags().StringVarP(&channelsArgs, "channels", "c", "", "The list of channels that bundle image belongs to")
-	if err := bundleBuildCmd.MarkFlagRequired("channels"); err != nil {
-		log.Fatalf("Failed to mark `channels` flag for `build` subcommand as required")
-	}
+	bundleBuildCmd.Flags().StringVarP(&channelsArgs, "channels", "c", "",
+		"The list of channels that bundle image belongs to" +
+		"(Required if `directory` is not pointing to a bundel in the nested bundel format)")
+
+	bundleBuildCmd.Flags().StringVarP(&imageBuilderArgs, "image-builder", "b", "docker",
+		"Tool to build container images. One of: [docker, podman, buildah]")
 
-	bundleBuildCmd.Flags().StringVarP(&imageBuilderArgs, "image-builder", "b", "docker", "Tool to build container images. One of: [docker, podman, buildah]")
+	bundleBuildCmd.Flags().StringVarP(&channelDefaultArgs, "default", "e", "",
+		"The default channel for the bundle image")
 
-	bundleBuildCmd.Flags().StringVarP(&channelDefaultArgs, "default", "e", "", "The default channel for the bundle image")
+	bundleBuildCmd.Flags().BoolVarP(&noChannelDefaultArgs, "no-default", "e", false,
```

Comment:
This seems like a not great user experience. We shouldn't add complexity here if it's not necessary. IMO we should just attempt to set the default channel like we were doing before if we can't find a packages.yaml file instead of asking the user if they want an error or not.

Reply:
? if we can not find the package.yaml then we have run the command. By then we would either successfully get the channel or that we didn't. If we got it we are writing it to the annotation file. If we didn't we would have already returned something. 

what do you mean by `then` we ask if users want error or not?  There isn't a second step command.

I can only think of returning a warning.

Reply:
Because before we didn't attempt to look for it and tried to guess it. Why not retain that implicitly without adding an extra parameter?